#! /bin/sh -e
##
## All lines beginning with `## DP:\' are a description of the patch.
## DP: Solves the FTBFS bugs. Updated for valgrind-2.2.0. 

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@

diff -urN valgrind-2.2.0.orig/coregrind/vg_unsafe.h valgrind-2.2.0/coregrind/vg_unsafe.h
--- valgrind-2.2.0.orig/coregrind/vg_unsafe.h	2004-08-29 21:02:28.000000000 +0000
+++ valgrind-2.2.0/coregrind/vg_unsafe.h	2004-09-01 23:28:42.000000000 +0000
@@ -54,7 +54,6 @@
 #include <linux/compiler.h> /* for __user definition */
 #endif
 #include <linux/msg.h>    /* for struct msgbuf */
-#include <asm/ipc.h>      /* for struct ipc_kludge */
 #include <linux/sem.h>    /* for struct sembuf */
 
 #include <scsi/sg.h>      /* for the SG_* ioctls */
diff -urN valgrind-2.2.0.orig/include/vg_kerneliface.h valgrind-2.2.0/include/vg_kerneliface.h
--- valgrind-2.2.0.orig/include/vg_kerneliface.h	2004-08-31 00:08:37.000000000 +0000
+++ valgrind-2.2.0/include/vg_kerneliface.h	2004-09-01 23:29:21.000000000 +0000
@@ -663,6 +663,13 @@
 #define VKI_MODIFY_LDT_CONTENTS_STACK       1
 #define VKI_MODIFY_LDT_CONTENTS_CODE        2
 
+/* IPC stuff, from 2.6.3 asm/ipc.h: */
+
+struct ipc_kludge {
+       struct msgbuf *msgp;
+       long msgtyp;
+};
+
 #define VKI_GDT_TLS_ENTRIES 3
 #define VKI_GDT_TLS_MIN     6
 #define VKI_GDT_TLS_MAX     (VKI_GDT_TLS_MIN + VKI_GDT_TLS_ENTRIES)
